### PR TITLE
BYOOS (non-Edge)

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR introduces the BYOOS concept and provides guidance on how users can create their own images and how to deploy those images to a host cluster. 

## Review Changes

💻 [Preview](https://deploy-preview-1257--docs-spectrocloud.netlify.app/cluster-profiles/byoos)
💻 [How-To Guide](https://deploy-preview-1257--docs-spectrocloud.netlify.app/cluster-profiles/byoos/image-builder)

🎫 [PCP-895](https://spectrocloud.atlassian.net/browse/PCP-895)
🎫 [PCP-800](https://spectrocloud.atlassian.net/browse/PCP-800)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1257#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 54c23f363a6cf5432b5993bbeaecaef9625011b1 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1654610022484676608) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | f138968cd9a13ed6a1807385f25bd0d4dac0cc35 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1654629907478614016) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/03-cluster-profiles/10-byoos.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+icon: ""
+hideToC: true
+fullWidth: false
+---
+
+# Overview
+
+With Palette, you can bring your own operating system and use it with your Kubernetes clusters using the [Bring Your Own Operating System (BYOOS)](/glossary-all#bringyourownoperatingsystem(byoos)) feature. The BYOOS pack allows you to upload your own OS images, configure the necessary drivers, and customize the OS to meet your specific requirements.
+
+Bringing your own operating system provides several benefits, including the ability to control your own dependencies, improve performance, and ensure compatibility with your existing applications. With BYOOS, you can choose the OS that best fits your needs, whether it's a commercial or open-source distribution, and easily integrate it with your Kubernetes clusters.
```
Comment (karl-cardenas-coding) [inline content/docs/03-cluster-profiles/10-byoos.md:14]:
```suggestion
Bringing your own operating system provides several benefits, including the ability to control your own dependencies, improve performance, and ensure compatibility with your existing applications. With BYOOS, you can choose the OS that best fits your needs, whether it's a commercial or open-source distribution, and integrate it with your Kubernetes clusters.
```

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 6348faf3a540a6953010dc5a6fc97cfdf0cb9d7f | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1654726042390106112) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 2baa1bc6b56d0afc62b0711ae11bb86b9a9803c0 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656361226801778688) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 6425b6407046813b4353241d88ab8b99ee9d4126 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656362233657036800) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 016a9d87c1bc287942e1f8a8c873c4bcf1be519c | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656363115299737600) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | bef701684d3c3a43e5c3a6bd849d736907066ce6 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656363115299737601) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | ef075bbf43fc6236e7c611b6719e77a8c1ffac34 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656406651118292992) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 6795239a4aa7f5b7e356b909f73204df56c38134 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656407783760728064) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/03-cluster-profiles/10-byoos/10-image-builder.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+</WarningBox>
+
+# Validation
+
+Use the following steps to validate your custom image.
+
+1. You can validate that the custom image is working correctly by deploying a compute instance in the respective infrastructure provider you created the image in using the custom image. Review the compute instance logs to learn more about the problems if you encounter any issues.
+
+
+2. Next, deploy a host cluster that uses the cluster profile you created containing the custom image. Verify the cluster is deployed correctly and without any issues. If you encounter any problems, review the event logs of the cluster to gain more details about the issue. Check out the [Deploy a Cluster](/clusters/public-cloud/deploy-k8s-cluster/) tutorial for additional guidance on deploying a host cluster.
```
Comment (rahulhazra97) [inline content/docs/03-cluster-profiles/10-byoos/10-image-builder.md:316]:
```suggestion
2. Next, deploy a host cluster that uses the cluster profile you created containing the custom image. Verify that the cluster is deployed correctly and without any issues. If you encounter any problems, review the event logs of the cluster to gain more details about the issue. Check out the [Deploy a Cluster](/clusters/public-cloud/deploy-k8s-cluster/) tutorial for additional guidance on deploying a host cluster.
```

File: content/docs/17-glossary-all.md [hunk 3]
```diff
@@ -28,16 +28,20 @@ App Profiles are templates created with preconfigured services required for Pale
...
 ## Chart Repositories
 
-Chart Repositories are public or private web servers that host Helm Charts. A few popular chart registries, such as Bitnami, are mapped into Palette by default. Users may add additional public or private chart repositories as an administration setting to leverage charts from those repositories.
+Chart Repositories are web servers, either public or private, that host Helm Charts. By default, Palette includes several popular chart registries such as Bitnami. As an administrator, you can add additional public or private chart repositories to leverage charts from those sources. This feature provides greater flexibility in managing and deploying applications, allowing you to access and use Helm Charts from various sources in your Palette environment
```
Comment (rahulhazra97) [inline content/docs/17-glossary-all.md:38]:
```suggestion
Chart Repositories are web servers, either public or private, that host Helm Charts. By default, Palette includes several popular chart registries, such as Bitnami. As an administrator, you can add additional public or private chart repositories to leverage charts from those sources. This feature provides greater flexibility in managing and deploying applications, allowing you to access and use Helm Charts from various sources in your Palette environment.

```

File: content/docs/17-glossary-all.md [hunk 4]
```diff
@@ -28,16 +28,20 @@ App Profiles are templates created with preconfigured services required for Pale
...
 ## Cloud Account
 
 Cloud Accounts are where access credentials are stored for public and private clouds. It is used by the system to provide new cluster infrastructure and cluster resources. Cloud account information is treated as sensitive data and fully encrypted using the tenant's unique encryption key.
 
 ## Cluster Mode
-Cluster Mode enables you to create, deploy, and manage Kubernetes clusters and applications. In Cluster Mode, you can deploy Kubernetes clusters to public cloud providers, on-prem datacenters, and on the edge. 
+Cluster Mode enables you to create, deploy, and manage Kubernetes clusters and applications. In Cluster Mode, you can deploy Kubernetes clusters to public cloud providers, on-prem data centers, and on the edge.
```
Comment (rahulhazra97) [inline content/docs/17-glossary-all.md:44]:
```suggestion
Cluster Mode enables you to create, deploy, and manage Kubernetes clusters and applications. In Cluster Mode, you can deploy Kubernetes clusters to public cloud providers, on-prem data centers, and on the Edge. 

```

**Review by rahulhazra97 — CHANGES_REQUESTED** [review]

File: content/docs/03-cluster-profiles/10-byoos.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+fullWidth: false
+---
+
+# Overview
+
+With Palette, you can bring your own operating system and use it with your Kubernetes clusters using the [Bring Your Own Operating System (BYOOS)](/glossary-all#bringyourownoperatingsystem(byoos)) feature. The BYOOS pack allows you to upload your own OS images, configure the necessary drivers, and customize the OS to meet your specific requirements.
+
+Bringing your own operating system provides several benefits, including the ability to control your own dependencies, improve performance, and ensure compatibility with your existing applications. With BYOOS, you can choose the OS that best fits your needs, whether it's a commercial or open-source distribution, and integrate it with your Kubernetes clusters.
+
+The BYOOS feature is especially useful for enterprises and organizations that have strict requirements around security, compliance, or specific hardware configurations. With the ability to bring your own OS, you can ensure that your Kubernetes clusters meet these requirements, without compromising on performance or functionality.
```
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply content/docs/03-cluster-profiles/10-byoos.md:16]: @rahulhazra97 Both Rita and feel that on should remain in this sentence.

File: content/docs/03-cluster-profiles/10-byoos/10-image-builder.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+  ![View of the cluster profile wizard](/clusters_byoos_image-builder_cluster-profile-byoos-yaml.png)
+
+
+17. Click on **Next layer** to add the Kubernetes layer.
+
+
+18. Select the desired Kubernetes distribution and version. Click on the **</\>** button to reveal the YAML editor.
+
+
+19. If you use a commercial OS, such as RHEL, you must activate the subscription. You can use the `preKubeadmCommands` section of the Kubernetes layer to issue commands before Kubernetes is started. In this example, the [RHEL Subscription Manager CLI](https://access.redhat.com/solutions/253273) is used to register the compute instance. Other OS may have different activation methods. Review the respective OS's documentation to learn more about activation methods.
```
Comment (karl-cardenas-coding) [inline content/docs/03-cluster-profiles/10-byoos/10-image-builder.md:280]:
```suggestion
19. If you use a commercial OS such as RHEL, you may have to activate the subscription. You can use the `preKubeadmCommands` section of the Kubernetes layer to issue commands before Kubernetes is started. In this example, the [RHEL Subscription Manager CLI](https://access.redhat.com/solutions/253273) is used to register the compute instance. Other OS may have different activation methods. Review the respective OS's documentation to learn more about activation methods.
```

File: content/docs/03-cluster-profiles/10-byoos/10-image-builder.md (hunk 2 above)
Comment (karl-cardenas-coding) [reply content/docs/03-cluster-profiles/10-byoos/10-image-builder.md:316]: I think this "that" is unwarranted in this case.

File: content/docs/17-glossary-all.md (hunk 3 above)
Comment (karl-cardenas-coding) [reply content/docs/17-glossary-all.md:38]: @rahulhazra97 In this case, the comma is not needed as we are ending the sentence immediately and not listing more than one item.

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 47dcf2696f0f2705ed8513fa6f9c7baf167374ce | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656724374499102720) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/03-cluster-profiles/10-byoos/10-image-builder.md [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+
+![A diagram displaying the steps for creating a custom image](/cluster-profiles_byoos_image-builder_workflow-diagram.png) <br />
+
+1. You will download the KIB project and configure the image builder's `packer.json` file.
+
+
+2. Use the `make` command to create a custom image containing a specific Operating System (OS) version and flavor.
+
+
+3. The custom image is created and distributed to the target regions you specified in the `packer.json` file.
```
Comment (karl-cardenas-coding) [inline content/docs/03-cluster-profiles/10-byoos/10-image-builder.md:33]:
```suggestion
3. The custom image is created and distributed to the target regions you specified in the **packer.json** file.
```

File: content/docs/03-cluster-profiles/10-byoos/10-image-builder.md [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+
+## Kubernetes Image Builder
+
+The Kubernetes Image Builder (KIB) is a project designed to help users create images for various platforms. The project is a consolidation of multiple tools that together work to create an artifact, or in simpler terms, a custom image. 
+
+You can use the custom images created by KIB with Palette, assuming the infrastructure provider is supported in Palette. Use the following diagram to understand how you can use KIB to create custom images that you can use with Palette.
+
+![A diagram displaying the steps for creating a custom image](/cluster-profiles_byoos_image-builder_workflow-diagram.png) <br />
+
+1. You will download the KIB project and configure the image builder's `packer.json` file.
```
Comment (karl-cardenas-coding) [inline content/docs/03-cluster-profiles/10-byoos/10-image-builder.md:27]:
```suggestion
1. You will download the KIB project and configure the image builder's **packer.json** file.
```

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | f1b55964cf277d76c15cdb7abbbe49f7ec563dd2 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1656726136039673856) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by rahulhazra97 — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1257#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1257#pullrequestreview-1423194232" title="Approved">rahulhazra97</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/17-glossary-all.md [hunk 9]
```diff
@@ -28,16 +28,20 @@ App Profiles are templates created with preconfigured services required for Pale
...
+For integrations and add-ons orchestrated via Palette [Packs](#pack) or [Charts](#helm-charts), at times it is required to provide additional Kubernetes resources to complete the installation. Resources like additional secrets or Custom Resource Definitions may need to be installed for the integration or add-on to function correctly. Attach Manifests are additional raw manifests attached to a cluster profile layer built using a Palette Pack or a Chart. Multiple Attach Manifests can be added to a layer in a cluster profile.
+
+## Bring Your Own Operating System (BYOOS)
+
+A feature in Palette that allows you to bring your own operating system and use it with your Kubernetes clusters. With the BYOOS pack, you can upload your own OS images, configure the necessary drivers, and customize the OS to meet your specific requirements. BYOOS gives you greater flexibility, control, and customization options when it comes to managing your Kubernetes clusters. It is especially useful for enterprises and organizations with strict requirements around security, compliance, or specific hardware configurations.
```
Comment (karl-cardenas-coding) [inline content/docs/17-glossary-all.md:35]:
```suggestion
A feature in Palette that allows you to bring your own operating system and use it with your Kubernetes clusters. With the BYOOS pack, you can reference your own OS images, configure the necessary drivers, and customize the OS to meet your specific requirements. BYOOS gives you greater flexibility, control, and customization options when it comes to managing your Kubernetes clusters. It is especially useful for enterprises and organizations with strict requirements around security, compliance, or specific hardware configurations.

```

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | dc67cc09a03b163dec5cf3c31539080246a274cc | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1257/librarium_build/1657131437788237824) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
